ocaml/libs/tracing/tracing.ml

```diff
@@ -314,7 +314,7 @@ module Spans = struct
                     let elapsed =
                       Unix.gettimeofday () -. span.Span.begin_time
                     in
-                    if elapsed > !span_timeout *. 1000000. then (
+                    if elapsed > !span_timeout then (
```

Comment:
This check is susceptible to clock changes, if the time spans are all created within xapi, it would be better to use [`Mtime_clock.elapsed`](https://ocaml.org/p/mtime/latest/doc/Mtime_clock/index.html#val-elapsed)

Reply:
I don't think we pass spans around, but only span context with its ids, and these are then used to create spans locally within each host. So local monotonic clock should be fine.

Reply:
But this is for _distributed_ tracing, where spans come from different processes and/or hosts, and are combined into a single trace. So we need wall-clock time on them.

Reply:
Yes I see a comment about the span begin_time "Using gettimeofday over Mtime as it is better for sharing timestamps between the systems". Would Ptime be better than gettimeofday though as a wall clock?

Would the clock changes matter much in the scale of a 24 hour timeout?

Reply:
Ptime and gettimeofday are the same, they are both wall clocks. Ptime just offers a nicer way to format the time as a string, for opentelemetry I think you need it as a float so there isn't much difference, 'gettimeofday' should be fine for that purpose.

Reply:
Yeah so I think the best solution is to add a separate monotonic timer for removing inactive spans as Rob suggests. I'll move this to a separate PR